app/Http/Controllers/api/StudentCollaborationController.php

```diff
@@ -3,20 +3,31 @@
 namespace App\Http\Controllers\api;
 
 use App\Http\Controllers\Controller;
+use App\Service\CollaborationService;
+use Exception;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
 
 //use Illuminate\Http\Request;
 
 class StudentCollaborationController extends Controller
 {
-    public function __invoke()
-    {
-
-        $data = file_get_contents(base_path('database/data/studentCollaborations.json'));
-
-        $decodedData = json_decode($data, true);
-
-        return response()->json($decodedData);
-
+    protected $collaborationService;
 
+    public function __construct(CollaborationService $collaborationService)
+    {
+        $this->collaborationService = $collaborationService;
+    }
+    public function __invoke($uuid)
+    {
+        try {
+            $collaborationDetail = [
+                'collaborations' => $this->collaborationService->getCollaborationDetails($uuid),
+            ];
+            return response()->json($collaborationDetail);
+        } catch (ModelNotFoundException $e) {
```

Comment:
You could use: 
catch (StudentNotFoundException $e) {
            return response()->json(['message' => $e->getMessage()], $e->getCode());